src/main/kotlin/uk/gov/justice/digital/hmpps/jobsboard/api/jobs/application/ExpressionOfInterestEditor.kt

```diff
@@ -0,0 +1,56 @@
+package uk.gov.justice.digital.hmpps.jobsboard.api.jobs.application
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.jobsboard.api.commons.domain.EntityId
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.ExpressionOfInterest
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.ExpressionOfInterestId
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.ExpressionOfInterestRepository
+import uk.gov.justice.digital.hmpps.jobsboard.api.jobs.domain.JobRepository
+
+@Service
+class ExpressionOfInterestEditor(
```

Comment:
I can see that you introduced the word 'Editor' on the application layer. For me, this layer represents the use cases of the application service, and I try to maintain them decoupled. The word 'editor' sounds too ambiguous to me (Can I update an expression of interest?)

I started to follow a simple rule for the use case naming. I used as the action verb the CRUD (Create, Retrieve, Update, Delete) options. The idea is to have grouped these classes by actions. The advantages that I can see are:

- Two developers can work on the update or with the retrieve in different branches, and they won't clash with each other
- You can see in one go the capabilities of each use case: I can create or update, and delete, but not retrieve.
- The classes stay small, and the cognitive load is reduced.

Reply:
noted, let me split up

Reply:
will split this

Reply:
updated